src/components/NoHouseholdIdModal.svelte

```diff
@@ -0,0 +1,55 @@
+<script lang="ts">
+import type { UpdatePolicyRequestBody } from 'data/policies'
+import { Dialog, setNotice, TextField } from '@silintl/ui-components'
+import { createEventDispatcher } from 'svelte'
+
+export let open = false
+
+const policyData: UpdatePolicyRequestBody = {}
+
+let title = 'Missing information'
+let householdId = ''
+let hasClosed: boolean = false
+
+const buttons: Dialog.AlertButton[] = [
+  { label: 'Go Back', action: 'cancel', class: 'mdc-dialog__button' },
+  { label: 'Submit', action: 'submit', class: 'mdc-button--raised' },
+]
+
+const dispatch = createEventDispatcher<{ closed: { choice: string; data?: UpdatePolicyRequestBody } }>()
+
+const handleDialog = (event: CustomEvent) => {
+  const choice: string = event.detail || ''
+  if (choice === 'submit') {
+    householdId = householdId.replaceAll(' ', '')
+    if (isIdValid(householdId)) {
+      policyData.household_id = householdId
+      dispatch('closed', { choice, data: policyData })
+      hasClosed = true
+    } else {
+      setNotice('Please enter a valid Household ID')
+    }
+  } else if (choice === 'cancel') {
+    dispatch('closed', { choice })
+    hasClosed = true
+    // prevents emiiting a second 'closed' event
+  } else if (!hasClosed) {
+    dispatch('closed', { choice })
+  }
+}
+
+const isIdValid = (sanitizedId: string): number | boolean =>
+  sanitizedId.length && sanitizedId.split('').every((digit) => /[0-9]/.test(digit))
```

Comment:
This might accomplish the "boolean only" goal:
```suggestion
  (sanitizedId.length > 0) && sanitizedId.split('').every((digit) => /[0-9]/.test(digit))
```